### ISSUE TITLE
Swift 3

### ISSUE THREAD

**Issue by SebastianKumor** [chiahsien/CHTCollectionViewWaterfallLayout#148]
How did you resolve this ins swift 3 ? 

 let yOffset = ((self.columnHeights[section] as AnyObject).object(columnIndex) as AnyObject).doubleValue 

gives me error 'Cannot call value of non-function type 'Any?!''

(self.columnHeights[section] as AnyObject).enumerateObjects({(object : AnyObject!, idx : NSInteger,pointer :UnsafeMutablePointer<ObjCBool>) and this one gives error ' Cannot invoke 'enumerateObjects' with an argument list of type '((AnyObject!, NSInteger, UnsafeMutablePointer<ObjCBool>) -> ())';

 I used Xcode 8 converter and this is the result, thx for help

**Comment by gsmeros** [comment]
Same here .. any updates ?

**Comment by SebastianKumor** [comment]
I did manage to resolve the errors but the `func longestColumnIndexInSection ()` is crashing m app

**Comment by MindaugasJucius** [comment]
```
func shortestColumnIndexInSection (_ section: NSInteger) -> NSInteger {
        var index = 0
        var shorestHeight = MAXFLOAT
        (self.columnHeights[section] as AnyObject).enumerateObjects { (obj: Any, idx: NSInteger, pointer: UnsafeMutablePointer<ObjCBool>) in
            let height = obj as! Float
            if (height<shorestHeight){
                shorestHeight = height
                index = idx
            }
        }
        return index
    }
```

**Comment by SebastianKumor** [comment]
@MindaugasJucius thx buddy, it seems like it working, tho I still have crash elsewhere

**Comment by chiahsien** [comment]
The Swift version is contributed by community, I personally don't adopt to Swift yet. So, welcome to send me a PR.

**Comment by SebastianKumor** [comment]
Ok guys thx, I managed to make it work

**Comment by gsmeros** [comment]
Hello Sebastian , can you please share your CHT file that works ? Thank you in advance.

**Comment by SebastianKumor** [comment]
@gsmeros @chiahsien  I updated the swift file in fork I created, it works for me https://github.com/SebastianKumor/CHTCollectionViewWaterfallLayout

**Comment by rlam3** [comment]
@SebastianKumor If it works for you. Please create a new cocoapods for the for the swift version since OP wont support swift. Or create a pull request for OP to create a new branch for? Thanks!

**Comment by chiahsien** [comment]
I've merged this PR https://github.com/chiahsien/CHTCollectionViewWaterfallLayout/pull/149
Please help to test if everything works as expected!

**Comment by SebastianKumor** [comment]
@rlam3 done, can you please also test it if it works as expected  ? thanks

**Comment by vikht0r** [comment]
@SebastianKumor,  it works :) . Thanks

**Comment by OviBortas** [comment]
I have an Issue when building this in Xcode 8.1, the layout does not take effect. It just lays everything out in one column.

**Comment by SebastianKumor** [comment]
@OviBortas same here, for me it crashes on `func longestColumnIndexInSection()`

**Comment by jvk75** [comment]
Hi, 
I created a pull #152 which makes the swift class more swifty and fixes issues with Xcode8.1

**Comment by SebastianKumor** [comment]
@jvk75 tested, and it works. @OviBortas try new update @jvk75  made

**Comment by chiahsien** [comment]
Hi all, I've merged #152 to `develop`, also I've merged #151 and #153.
Could anyone help to test if the `develop` branch works as expected?

**Comment by SebastianKumor** [comment]
Hi @chiahsien , I tested this on xcode 8.1 and swift 3.0.1 and it works well.
